src/stores/DetailsForm/Store.ts

```diff
@@ -295,14 +318,56 @@ export const getInitialState = (
             workflow === 'materialization_create';
 
         if (connectorId) {
-            const { setHydrationErrorsExist } = get();
+            const {
+                setDataPlaneOptions,
+                setHydrationError,
+                setHydrationErrorsExist,
+            } = get();
+
+            let dataPlaneOptions: DataPlaneOption[] | null =
+                dataPlaneFeatureFlag === 'show_option' ? [] : null;
+
+            if (dataPlaneOptions !== null) {
+                const dataPlaneResponse = await getDataPlaneOptions();
+
+                if (
+                    !dataPlaneResponse.error &&
+                    dataPlaneResponse.data &&
+                    dataPlaneResponse.data.length > 0
+                ) {
+                    dataPlaneOptions = dataPlaneResponse.data.map(
+                        ({ data_plane_name, id }) => {
+                            const scope = getDataPlaneScope(data_plane_name);
+
+                            const { cluster, prefix, provider, region } =
+                                parseDataPlaneName(data_plane_name, scope);
+
+                            return {
+                                dataPlaneName: {
+                                    cluster,
+                                    prefix,
+                                    provider,
+                                    region,
+                                    whole: data_plane_name,
```

Comment:
I think we should just put that inside the `parseDataPlaneName` function. Even though it is kinda weird that it is just passing back what was provided... but then we wouldn't need the destruct up above and we have a "one stop shop" for name related stuff.

Reply:
Agreed.